presto-resource-group-managers/src/test/java/com/facebook/presto/resourceGroups/db/TestDbManagerSpecProvider.java

```diff
@@ -0,0 +1,173 @@
+
+        // when we load the selectors we expect the selector list to be ordered by priority
+        expectedUsers.sort(Comparator.<String>comparingInt(Integer::parseInt).reversed());
+
+        for (int i = 0; i < numberOfUsers; i++) {
+            Optional<Pattern> user = selectors.get(i).getUserRegex();
+            assertTrue(user.isPresent());
+            assertEquals(user.get().pattern(), expectedUsers.get(i));
+        }
+    }
+
+    @Test
+    public void testDuplicates()
+    {
+        H2DaoProvider daoProvider = setup("test_dup_roots");
+        H2ResourceGroupsDao dao = daoProvider.get();
+        dao.createResourceGroupsGlobalPropertiesTable();
+        dao.createResourceGroupsTable();
+        dao.createSelectorsTable();
+        dao.insertResourceGroup(1, "global", "1MB", 1000, 100, 100, null, null, null, null, null, null, ENVIRONMENT);
+        try {
+            dao.insertResourceGroup(1, "global", "1MB", 1000, 100, 100, null, null, null, null, null, null, ENVIRONMENT);
+            fail("Expected to fail");
+        }
+        catch (RuntimeException ex) {
+            assertTrue(ex instanceof UnableToExecuteStatementException);
+            assertTrue(ex.getCause() instanceof JdbcSQLIntegrityConstraintViolationException);
+            assertTrue(ex.getCause().getMessage().startsWith("Unique index or primary key violation"));
+        }
+        dao.insertSelector(1, 1, null, null, null, null, null);
+        daoProvider = setup("test_dup_subs");
+        dao = daoProvider.get();
+        dao.createResourceGroupsGlobalPropertiesTable();
+        dao.createResourceGroupsTable();
+        dao.createSelectorsTable();
+        dao.insertResourceGroup(1, "global", "1MB", 1000, 100, 100, null, null, null, null, null, null, ENVIRONMENT);
+        dao.insertResourceGroup(2, "sub", "1MB", 1000, 100, 100, null, null, null, null, null, 1L, ENVIRONMENT);
+        try {
+            dao.insertResourceGroup(2, "sub", "1MB", 1000, 100, 100, null, null, null, null, null, 1L, ENVIRONMENT);
+        }
+        catch (RuntimeException ex) {
+            assertTrue(ex instanceof UnableToExecuteStatementException);
+            assertTrue(ex.getCause() instanceof JdbcSQLIntegrityConstraintViolationException);
+            assertTrue(ex.getCause().getMessage().startsWith("Unique index or primary key violation"));
+        }
+
+        dao.insertSelector(2, 2, null, null, null, null, null);
+    }
+
+    @Test
+    public void testSubgroups()
+    {
+        H2DaoProvider daoProvider = setup("test_dup_roots");
+        H2ResourceGroupsDao dao = daoProvider.get();
+        dao.createResourceGroupsGlobalPropertiesTable();
+        dao.createResourceGroupsTable();
+        dao.createSelectorsTable();
+        dao.insertResourceGroup(1, "global", "1MB", 1000, 100, 100, null, null, null, null, null, null, ENVIRONMENT);
+        dao.insertResourceGroup(2, "sub", "1MB", 1000, 100, 100, null, null, null, null, null, 1L, ENVIRONMENT);
```

Comment:
let's add one more subgroup to the root.
and another level.
Something like this
1 -> 2,3
2 -> 4, 5
3 -> 6